File: minji/week1/10828.cpp
/*
	[BOJ] 10828 Ω∫≈√
*/
#include <iostream>
#include <stack>
using namespace std;

int main() {
	stack<int> s;
	int N = 0;
	cin >> N;
	while (N--) {
		string cmd;
		int num = 0;
		cin >> cmd;
		if (cmd == "push") {
			cin >> num;
			s.push(num);
		}
		else if (cmd == "pop") {
			if (s.empty() == 1) {
				cout << "-1\n";
			}
			else {
				cout<<s.top()<<"\n";
				s.pop();
			}
		}
		else if (cmd == "size") {
			cout << s.size()<<"\n";
		}
		else if (cmd == "empty") {
			cout << s.empty() << "\n";
		}
		else if (cmd == "top") {
			if (s.empty() == 1) {
				cout << "-1\n";
			}
			else {
				cout << s.top() << "\n";
			}
		}
	}
	return 0;
}

File: minji/week1/10828.py
'''
    [BOJ] 10828 스택
'''
import sys
N = int(sys.stdin.readline().rstrip())
s = []
for _ in range(0,N):
    cmd = sys.stdin.readline().rstrip().split()
    if cmd[0] == "push":
        s.append(cmd[1])
    elif cmd[0] == "pop":
        #if len(s) == 0:
        if (not s):
            print(-1)
        else:
            print(s.pop())
    elif cmd[0] == "size":
        print(len(s))
    elif cmd[0] == "empty":
        #if len(s) == 0:
        if (not s):
            print(1)
        else:
            print(0)
    elif cmd[0] == "top":
        if s:
            print(s[-1])
        else:
            print(-1)



